herbert_core/utilities/classes/@unique_objects_container/private/add_single_.m

```diff
@@ -1,12 +1,29 @@
-function [self,nuix] = add_single_(self,obj)
+function [self,nuix] = add_single_(self,obj,ix,hash)
 %ADD_SINGLE_ Add single object to the unique objects container
 
 
 % Find if the object is already in the container. ix is
 % returned as the index to the object in the container.
 % hash is returned as the hash of the object. If ix is empty
 % then the object is not in the container.
-[ix,hash] = self.find_in_container(obj);
+%
+% Input
+% -----
+% self - the unique_objects_container in question
+% obj  - the object to be added to the container
+% ix   - the unique index of the object if it is already in the container. 
+%        this will have been found from a previous call of find_in_container
+% hash - the hash for obj previously made by that call to find_in_container
+% 
+% Output
+% ------
+% self - the modified container (modified by adding obj)
+% nuix - the insertion index at which obj is added in the container
+
+% if ix and hash are not specified, call find_in_container to get them
+if nargin<=2
```

Comment:
Suggestion:
I prefer not using `nargin` to check arguments as it limits future function signature changes. Instead, prefer explicitly checking `exist('variable', 'var')`